### PR TITLE
Complete User Stories 23-24: increment/decrement links to change cart item quantity

### LINKED ISSUE #25: User Story 23, Adding Item Quantity to Cart
As a visitor
When I have items in my cart
And I visit my cart
Next to each item in my cart
I see a button or link to increment the count of items I want to purchase
I cannot increment the count beyond the item's inventory size

### LINKED ISSUE #26: User Story 24, Decreasing Item Quantity from Cart
As a visitor
When I have items in my cart
And I visit my cart
Next to each item in my cart
I see a button or link to decrement the count of items I want to purchase
If I decrement the count to 0 the item is immediately removed from my cart

### PR DESCRIPTION
**Pull request by DanielEFrampton**
## Description
Resolves #25 
Resolves #26 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## RSpec Results
```
Finished in 5.21 seconds (files took 4.15 seconds to load)
134 examples, 0 failures

Coverage report generated for RSpec to /Users/dframpton/turing/2module/grou
p_projects/monster_shop_part_1/coverage. 1769 / 1769 LOC (100.0%) covered.
```

### CODE REVIEW

**Review by philjdelong — COMMENTED** [review]
Testing is on point here Mr. Frampton. 

Excellent work.

File: Gemfile.lock [hunk 1]
```diff
@@ -104,7 +104,7 @@ GEM
       method_source (~> 0.9.0)
     public_suffix (4.0.1)
     puma (3.12.2)
-    rack (2.0.7)
+    rack (2.0.8)
```
Comment (KirbyDD) [inline Gemfile.lock:107]: what does this do?

Comment (DanielEFrampton) [reply Gemfile.lock:107]: Don't know, I think it happened when I ran `bundle update`

File: app/models/cart.rb [hunk 2]
```diff
@@ -31,20 +31,20 @@ def total
...
+    @contents[item_id] -= 1
+  end
+
+  def limit_reached?(item_id)
+    @contents[item_id] == Item.find(item_id).inventory
+  end
+  
+  def quantity_zero?(item_id)
+    @contents[item_id] == 0
+  end
```
Comment (KirbyDD) [inline app/models/cart.rb:49]: great work on these methods  😹

File: app/views/cart/show.html.erb [hunk 3]
```diff
@@ -29,10 +29,14 @@
...
-          <p><%=quantity%></p>
+        <span id="cart-item-<%=item.id%>-quantity">
+          <p>
+            <%= link_to '-', "/cart?increment_decrement=decrement&item_id=#{item.id}", method: :patch %>
+            <%= quantity %>
+            <%= link_to '+', "/cart?increment_decrement=increment&item_id=#{item.id}", method: :patch %>
+          </p>
           <p><%= link_to "Remove", "/cart/#{item.id}",method: :delete %></p>
-        </center>
+        </span>
```
Comment (KirbyDD) [inline app/views/cart/show.html.erb:39]: you only use span's as an inline targeting element and it shouldnt be used as a container

**Review by KirbyDD — APPROVED** [review]
That span while it wont technically break anything needs to be switched for a proper tag

**Review by mel-rob — APPROVED** [review]
Looks awesome. Thanks for working on this!

File: app/models/cart.rb (hunk 2 above)
Comment (DanielEFrampton) [reply app/models/cart.rb:49]: Thank you. Thank you very much

**Pull request merged**